node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/Status.java

```diff
@@ -96,26 +100,36 @@ public boolean wantToDeprovision() {
     }
 
     public Status withHardwareDivergence(Optional<String> hardwareDivergence) {
-        return new Status(reboot, vespaVersion, failCount, hardwareFailureDescription, wantToRetire, wantToDeprovision, hardwareDivergence, osVersion);
+        return new Status(reboot, vespaVersion, failCount, hardwareFailureDescription, wantToRetire, wantToDeprovision, hardwareDivergence, osVersion, firmwareVerifiedAt);
     }
 
     /** Returns hardware divergence report as JSON string, if any */
     public Optional<String> hardwareDivergence() { return  hardwareDivergence; }
 
     /** Returns a copy of this with the current OS version set to version */
     public Status withOsVersion(Version version) {
-        return new Status(reboot, vespaVersion, failCount, hardwareFailureDescription, wantToRetire, wantToDeprovision, hardwareDivergence, Optional.of(version));
+        return new Status(reboot, vespaVersion, failCount, hardwareFailureDescription, wantToRetire, wantToDeprovision, hardwareDivergence, Optional.of(version), firmwareVerifiedAt);
     }
 
     /** Returns the current OS version of this node, if any */
     public Optional<Version> osVersion() {
         return osVersion;
     }
 
+    /** Returns a copy of this with the firmwareVerifiedAt set to the given instant. */
+    public Status withFirmwareVerifiedAt(Instant instant) {
```

Comment:
Is there any reason this can't just be a generation, like restart/reboot?

Reply:
`Generation` is meant for strictly increasing sequences, and does not support reset or cancellation.
Here, we want to be able to cancel a request for a check. 
Also, I think it's nice to be able to read when the firmware was last checked. 

Are there any reasons you don't think a timestamp is a good idea?